homeassistant/components/aprilaire/sensor.py

```diff
@@ -0,0 +1,309 @@
+    ),
+)
+
+
+class BaseAprilaireSensor(BaseAprilaireEntity, SensorEntity):
+    """Base sensor entity for Aprilaire."""
+
+    entity_description: AprilaireSensorDescription
+    status_sensor_available_value: int | None = None
+    status_sensor_exists_values: list[int] | None = None
+
+    def __init__(
+        self,
+        coordinator: AprilaireCoordinator,
+        description: AprilaireSensorDescription,
+        unique_id: str,
+    ) -> None:
+        """Initialize a sensor for an Aprilaire device."""
+
+        self.entity_description = description
+
+        super().__init__(coordinator, unique_id)
+
+    @property
+    def exists(self) -> bool:
+        """Return True if the sensor exists."""
+
+        if (
+            self.entity_description.status_key is None
+            or self.status_sensor_exists_values is None
+        ):
+            return True
+
+        return (
+            self.coordinator.data.get(self.entity_description.status_key)
+            in self.status_sensor_exists_values
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if the sensor is available."""
+
+        if (
+            self.entity_description.status_key is None
+            or self.status_sensor_available_value is None
+        ):
+            return True
+
+        if not super().available:
+            return False
+
+        return (
+            self.coordinator.data.get(self.entity_description.status_key)
+            == self.status_sensor_available_value
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor."""
+
+        # Valid cast as pyaprilaire only provides str | int | float
+        return cast(
+            StateType, self.coordinator.data.get(self.entity_description.value_key)
+        )
+
+
+class AprilaireHumiditySensor(BaseAprilaireSensor):
+    """Humidity sensor entity for Aprilaire."""
+
+    status_sensor_available_value = 0
+    status_sensor_exists_values = [0, 1, 2]
+
+
+class AprilaireTemperatureSensor(BaseAprilaireSensor):
+    """Temperature sensor entity for Aprilaire."""
+
+    status_sensor_available_value = 0
+    status_sensor_exists_values = [0, 1, 2]
+
+    @property
+    def suggested_display_precision(self) -> int | None:
+        """Return the suggested number of decimal digits for display."""
+        if self.unit_of_measurement == UnitOfTemperature.CELSIUS:
+            return 1
+
+        return 0
+
+
+class AprilaireStatusSensor(BaseAprilaireSensor):
+    """Status sensor entity for Aprilaire."""
+
+    status_sensor_exists_values = [1, 2]
+    entity_description: AprilaireStatusSensorDescription
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor mapped to the status option."""
+
+        raw_value = super().native_value
+
+        return self.entity_description.status_map.get(cast(int, raw_value))
```

Comment:
Instead of the cast here, we can declare the type of the status maps as `dict[StateType, str]` to show that we intend to index them with any `StateType`

Reply:
Done